solr/solrj/src/java/org/apache/solr/client/solrj/request/CollectionAdminRequest.java

```diff
@@ -446,7 +444,6 @@ public static Modify modifyCollection(String collection, Map<String, Object> pro
     protected Boolean autoAddReplicas;
     protected String alias;
     protected String[] rule , snitch;
-    protected String withCollection;
```

Comment:
Do we remove the notion of `withCollection` with the Autoscaling framework?
I would prefer to let the client manage it manually (i.e. specify its replica placement carefully) if there's a co-location need, but if we keep it we need to keep it here as well I believe, and if we remove it there are many other places from which it should be removed.

Reply:
`withCollection` has made an already complex rule engine even more complex.

Reply:
Removed withCollection support. We can see if this can be implemented in an elegant way going forward. Also, with XCJF thing, this may now be irrelevant.